### PR TITLE
MAINT: unskip test on win32

### PR DESCRIPTION
**Pull request by charris**
Backport of #15187. 

Continuation of gh-15170; following the suggestion of @eric-wieser

Remove `abspath()`, enable tests on win32

### CODE REVIEW

**Comment by charris** [comment]
The test failure looks valid and only on Python 3.5.

**Comment by mattip** [comment]
The problem is [bpo-26027](https://bugs.python.org/issue27524). xref gh-15225

File: numpy/random/tests/test_extending.py [hunk 1]
```diff
@@ -40,10 +40,8 @@
 
 @pytest.mark.skipif(cython is None, reason="requires cython")
 @pytest.mark.slow
-@pytest.mark.skipif(sys.platform == 'win32', reason="cmd too long on CI")
 def test_cython(tmp_path):
     examples = os.path.join(os.path.dirname(__file__), '..', '_examples')
-    base = os.path.dirname(examples)
     shutil.copytree(examples, tmp_path / '_examples')
```
Comment (mattip) [inline numpy/random/tests/test_extending.py]: This should become `str(tmp_path / '_examples')`

Comment (charris) [reply numpy/random/tests/test_extending.py]: OK, will make the change.

**Review by mattip — APPROVED** [review]
Thanks, LGTM

**Pull request merged**